### DISCUSSION TITLE
learning editor.js, why renderAction() disappears

### DISCUSSION THREAD

**Discussion by Florian418** [codex-team/editor.js#2862] [Q&A] [answered]
Hi, I'm trying to understand how editor.js works. I've been following the documentation on 'creating a block tool' and have now moved on to the section about 'Creating an Inline Tool.' To learn more, I’m studying this [editorjs-hyperlink:](https://github.com/trinhtam/editorjs-hyperlink/blob/master/src/Hyperlink.js)

However, I'm running into an issue with renderActions(). When I click on the inline button, the renderActions() implementation briefly appears but then immediately disappears. This started happening as soon as I began writing code in the surround() method, but I don't understand why this is happening and how to fix it.

https://github.com/user-attachments/assets/4f15962d-0230-4b3f-b4a9-52ee2417d565

And an example "marker button" here: [demo](https://nolan.ovh/need-help-with-editorjs-01)
The source code: [code](https://nolan.ovh/assets/libs/editorjs/flow_plugins/marker-tool.mjs)

Thank you for your help.

**Comment by Florian418** [comment, marked as answer]
Ok, I understand that I'm not managing the event listener correctly. I will work on this. Sorry for the inconvenience.

**Reply by Temmermans** [reply]
Hey @Florian418 Im having the same issue. How did you resolve this?

**Reply by Tomsgu** [reply]
The documentation is not updated to the new version I suppose, because it's working exactly as it's in the video. I am trying to understand what is the problem too. I wish you marked the response with a solution.

**Comment by chaegumi** [comment]
I may know the problem, checkState needs to return a value

```
    checkState(){
        const mark = this.api.selection.findParentTag(this.tag);

        this.state = !!mark;
        // console.log(this.state);
        if(this.state){
            this.showActions(mark);
            // this.colorPicker.style.display = 'block';
        }else{
            // this.hideActions();
        }

        return this.state; // important
    }
```
<img width="1226" height="823" alt="image" src="https://github.com/user-attachments/assets/eddc117c-7e94-4398-8185-dcf6e1105480" />
<img width="1095" height="406" alt="image" src="https://github.com/user-attachments/assets/b10b194a-c1d6-4323-b5a1-9d4f057bf8af" />

https://github.com/codex-team/editor.js/blob/next/docs/tools-inline.md#return-value-2
<img width="832" height="414" alt="image" src="https://github.com/user-attachments/assets/7fbd429c-8eda-41f2-870d-877c8be11106" />

**Reply by Tomsgu** [reply]
Oh I completely missed the (version: next) documentation and of course I didn't realized that change in the link tool either. 

Thanks for pointing that for us!
